apps/google-analytics-4/frontend/src/components/config-screen/map-account-property/MapAccountPropertySection.tsx

```diff
@@ -1,45 +1,61 @@
 import { useState, useEffect } from 'react';
-import {
-  Stack,
-  Box,
-  Subheading,
-  Select,
-  Spinner,
-  Paragraph,
-  TextLink,
-} from '@contentful/f36-components';
+import { Stack, Box, Subheading, Spinner, Paragraph, TextLink } from '@contentful/f36-components';
 import { AccountSummariesType } from 'types';
 import { KeyValueMap } from '@contentful/app-sdk/dist/types/entities';
 import { ExternalLinkIcon } from '@contentful/f36-icons';
+import MapAccountPropertyDropdown from 'components/config-screen/map-account-property/MapAccountPropertyDropdown';
 
 interface Props {
   accountsSummaries: AccountSummariesType[];
   parameters: KeyValueMap;
   mergeSdkParameters: Function;
   onIsValidAccountProperty: Function;
+  originalPropertyId: string;
+  isApiAccessLoading: boolean;
 }
 
-const getIdOnly = (unformattedId: string) => {
-  return unformattedId.split('/')[1];
-};
-
 export default function MapAccountPropertySection(props: Props) {
-  const { accountsSummaries, parameters, mergeSdkParameters, onIsValidAccountProperty } = props;
+  const {
+    accountsSummaries,
+    parameters,
+    mergeSdkParameters,
+    onIsValidAccountProperty,
+    originalPropertyId,
+    isApiAccessLoading,
+  } = props;
 
   const [selectedPropertyId, setSelectedPropertyId] = useState<string>('');
   const [sortedAccountSummaries, setSortedAccountSummaries] = useState<AccountSummariesType[]>([]);
   const [loadingProperties, setLoadingProperties] = useState<boolean>(true);
   const [loadingParameters, setLoadingParameters] = useState<boolean>(true);
+  const [isPropertyIdInOptions, setIsPropertyIdInOptions] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (parameters.propertyId === originalPropertyId) {
+      const isInOptions = sortedAccountSummaries.some((accountSummary) => {
+        return accountSummary.propertySummaries.some((propertySummary) => {
+          return propertySummary.property === originalPropertyId;
+        });
+      });
+
+      setIsPropertyIdInOptions(isInOptions);
+    }
+  }, [sortedAccountSummaries, originalPropertyId, parameters.propertyId]);
 
   useEffect(() => {
     if (parameters.propertyId) {
       setSelectedPropertyId(parameters.propertyId);
-      onIsValidAccountProperty(true);
+      // Prevent saving if selected property id is not in list of available properties
```

Comment:
Hm, do we want to do this?

I'm thinking that we might prefer to do what we do for content types -- that is leave them in an invalid state (withv visible error state) but allow you to save regardless. Keep in mind this would block you from saving other important changes you make to content types. It also might take some time for someone to get access, so they could be blocked for a while.

To be a bit more clear we could allow them to keep their existing value as is (even though it's not in the list) and save would go through.

WDYT?

Reply:
This is actually how the content type configuration is working -- if you have an invalid selection (the content type or the slug field that you had previously selected was deleted), then you are not able to save. My initial intention was to encourage the user to immediately address anything invalid, but your point does make sense. There are so many things on this configuration page, so it might not make sense to block a user from saving other parts of the configuration. I'll remove this behavior on the property id dropdown and will also add a note to remove that behavior in the task to clean up some UX with the content type selection. Thanks for pointing this out!